### PR TITLE
brigand: conan v2 support + add with_boost option

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — APPROVED** [review]

File: recipes/brigand/all/conanfile.py [hunk 1]
```diff
@@ -13,29 +14,51 @@ class BrigandConan(ConanFile):
...
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "with_boost": [True, False],
+    }
+    default_options = {
+        "with_boost": True,
+    }
+
     no_copy_sources = True
```
Comment (Tradias) [inline recipes/brigand/all/conanfile.py:26]: Is this correct. In all other conanfiles this is called `no_copy_source`?

**Review by toge — APPROVED** [review]

File: recipes/brigand/all/conanfile.py [hunk 2]
```diff
@@ -13,29 +14,51 @@ class BrigandConan(ConanFile):
...
-        self.cpp_info.names["pkg_config"] = "libbrigand"
+        self.cpp_info.set_property("pkg_config_name", "libbrigand")
+        self.cpp_info.bindirs = []
+        self.cpp_info.frameworkdirs = []
+        self.cpp_info.libdirs = []
+        self.cpp_info.resdirs = []
+        if self.options.with_boost:
+            self.cpp_info.requires = ["boost::headers"]
+        else:
+            self.cpp_info.defines("BRIGAND_NO_BOOST_SUPPORT")
```
Comment (Tradias) [inline recipes/brigand/all/conanfile.py:64]: Is this syntax correct? All other conanfiles say `self.cpp_info.defines = ["BRIGAND_NO_BOOST_SUPPORT"]`

Comment (SpaceIm) [reply recipes/brigand/all/conanfile.py:64]: good catch

File: recipes/brigand/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/brigand/all/conanfile.py]: Indeed, it was incorrect from the beginning

**Review by uilianries — APPROVED** [review]
LGTM

**Review by jgsogo — APPROVED** [review]

**Pull request merged**